x-pack/plugins/cases/public/components/filter_popover/index.test.tsx

```diff
@@ -110,4 +110,90 @@ describe('FilterPopover ', () => {
 
     expect(onSelectedOptionsChanged).toHaveBeenCalledWith([]);
   });
+
+  describe('maximum limit', () => {
+    const newTags = ['coke', 'pepsi', 'sprite', 'juice', 'water'];
+    const maxLength = 3;
+    const maxLengthLabel = `You have selected maximum number of ${maxLength} tags to filter`;
+
+    it('should show message when maximum options are selected', async () => {
+      const { getByTestId } = appMockRender.render(
+        <FilterPopover
+          buttonLabel={'Tags'}
+          onSelectedOptionsChanged={onSelectedOptionsChanged}
+          selectedOptions={[...newTags.slice(0, 3)]}
+          options={newTags}
+          optionsMaxLEngth={maxLength}
+          optionsMaxLengthLabel={maxLengthLabel}
+        />
+      );
+
+      userEvent.click(getByTestId('options-filter-popover-button-Tags'));
+
+      await waitForEuiPopoverOpen();
+
+      expect(getByTestId('maximum-length-warning')).toHaveTextContent(maxLengthLabel);
+
+      expect(getByTestId(`options-filter-popover-item-${newTags[4]}`)).toHaveProperty('disabled');
+    });
+
+    it('should not show message when maximum length label is missing', async () => {
+      const { getByTestId, queryByTestId } = appMockRender.render(
+        <FilterPopover
+          buttonLabel={'Tags'}
+          onSelectedOptionsChanged={onSelectedOptionsChanged}
+          selectedOptions={[newTags[0], newTags[2]]}
+          options={newTags}
+          optionsMaxLEngth={maxLength}
+        />
+      );
+
+      userEvent.click(getByTestId('options-filter-popover-button-Tags'));
+
+      await waitForEuiPopoverOpen();
+
+      expect(queryByTestId('maximum-length-warning')).not.toBeInTheDocument();
+      expect(getByTestId(`options-filter-popover-item-${newTags[4]}`)).toHaveProperty('disabled');
+    });
+
+    it('should not show message and disable options when maximum length property is missing', async () => {
+      const { getByTestId, queryByTestId } = appMockRender.render(
+        <FilterPopover
+          buttonLabel={'Tags'}
+          onSelectedOptionsChanged={onSelectedOptionsChanged}
+          selectedOptions={[newTags[0], newTags[2]]}
+          options={newTags}
```

Comment:
nit: Maybe we can add the `optionsMaxLengthLabel` to be sure that it does not affect it.